pkg/gcrcleaner/cleaner.go

```diff
@@ -160,6 +161,6 @@ func (c *Cleaner) deleteOne(ref gcrname.Reference) error {
 
 // shouldDelete returns true if the manifest has no tags or allows deletion of tagged images
 // and is before the requested time.
-func (c *Cleaner) shouldDelete(m gcrgoogle.ManifestInfo, since time.Time, allowTag bool) bool {
-	return (allowTag || len(m.Tags) == 0) && m.Uploaded.UTC().Before(since)
+func (c *Cleaner) shouldDelete(m gcrgoogle.ManifestInfo, since time.Time, allowTag bool, tagFilterRegexp regexp.Regexp) bool {
```

Comment:
Let's accept the pointer here instead of dereferencing everywhere else.
```suggestion
func (c *Cleaner) shouldDelete(m gcrgoogle.ManifestInfo, since time.Time, allowTag bool, tagFilterRegexp *regexp.Regexp) bool {
```

Reply:
Accepted on the latest commit.